src/jit/hwintrinsicxarch.cpp

```diff
@@ -337,44 +467,87 @@ GenTree* Compiler::impX86HWIntrinsic(NamedIntrinsic        intrinsic,
     {
         return gtNewIconNode(issupported);
     }
-    else if (!issupported)
+    // - calling to unsupported intrinsics must throw PlatforNotSupportedException
+    // - 64-bit (r64) intrinsics must throw PlatforNotSupportedException on 32-bit platform
+    else if (!issupported || is64bitOnlyIntrinsicOn32bit(category, callType))
     {
         return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
     }
 
+    // table-driven importer of simple intrinsics
+    if (impTableDrivableHWIntrinsic(category))
+    {
+        unsigned int sizeBytes;
+        var_types    baseType = getBaseTypeAndSizeOfSIMDType(sig->retTypeSigClass, &sizeBytes);
+        assert(baseType != TYP_UNKNOWN && sizeBytes != 0);
+        var_types retType  = getSIMDTypeForSize(sizeBytes);
+        unsigned  simdSize = simdSizeOfHWIntrinsic(intrinsic, sig);
+
+        assert(numArgs > 0);
+        assert(retType != TYP_UNDEF);
+        assert(retType == TYP_SIMD16 || retType == TYP_SIMD32);
+        assert(instructionOfHWIntrinsicOnType(intrinsic, baseType) != INS_invalid);
+        assert(simdSize == 32 || simdSize == 16);
+
+        GenTree* retNode = nullptr;
+        GenTree* op1     = nullptr;
+        GenTree* op2     = nullptr;
+
+        switch (numArgs)
+        {
+            case 1:
+                op1     = impPopStack().val;
+                retNode = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, baseType, simdSize);
+                break;
+            case 2:
+                op2     = impPopStack().val;
+                op1     = impPopStack().val;
+                retNode = gtNewSimdHWIntrinsicNode(retType, op1, op2, intrinsic, baseType, simdSize);
+                break;
+
+            default:
+                assert(numArgs >= 3);
+                op1     = buildArgList(numArgs);
+                retNode = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, baseType, simdSize);
+                break;
+        }
+        return retNode;
+    }
+
+    // other intrinsics need special importation
     switch (isa)
     {
         case InstructionSet_SSE:
-            return impSSEIntrinsic(intrinsic, method, sig, mustExpand);
+            return impSSEIntrinsic(intrinsic, method, sig);
```

Comment:
I don't think you can get rid of the `mustExpand` parameter on these methods, as that's what determines whether it can return `nullptr` so that it instead generates a call.